yb-voyager/cmd/migration_complexity.go

```diff
@@ -200,3 +210,121 @@ func getComplexityForLevel(level string, count int) string {
 		panic(fmt.Sprintf("unknown impact level %s for determining complexity", level))
 	}
 }
+
+// ======================================= Migration Complexity Explanation ==========================================
+
+const explainTemplateHTML = `
```

Comment:
Would like to avoid having HTML in multiple places. It should all ideally be in the template file. Morevoer, field `AssessmentReport.MigrationComplexityExplanation` having HTML would be weird. 
But I get why you did this. (because json and HTML have different requirements). 

Here's one way to do this: 
1. in `AssessmentReport`, instead of storing just the final string in `MigrationComplexityExplanation`, store the actual `ExplanationData` struct. 
2. in HTML template, you have all the necessary information to build the table there. 
3. for json, just exclude the summaries from it. (in fact there's no harm in even keeping it tbh).

Reply:
Agree to this, we should keep all the HTML code in that template 

> for json, just exclude the summaries from it. (in fact there's no harm in even keeping it tbh).
yes we can keep it in JSON as well an, in case of tests and all you are anyways ignoring it

Reply:
added a TODO comment